### DISCUSSION TITLE
uc_driver, download speed

### DISCUSSION THREAD

**Discussion by MickeyM007** [MacielG1/VodRecovery#1] [Q&A]
Hello,
I have a few questions about v1.0.2.
-On first start it automatically downloaded and installed Chrome drivers or something? `(uc_driver.exe)` Is it necessary? 
-Today I got an error code: `"Initialization of the cloud management controller aborted because CBCM is not enabled. Use --enable chrome-browser-cloud-management".` Chrome is not installed on my PC.
-The script works much faster and better, but is there a way to speed up the download speed? `The average now: speed=3.41x`.
  With the original script: `The average: speed=34.43x`.
-Sometimes I get:  `An error occurred: time data '21 Feb 2024 23:05:00' does not match format '%Y-%m-%d %H:%M:%S'
  Press Enter to exit.`

Thank you!!!

**Comment by MacielG1** [comment]
- You don't need to have Chrome installed.
- Did the error "Initialization of the cloud..." caused the program to stop? because it should continue even if that message is displayed. 
- The download speed should be more normal in the 1.0.3, but still requires more testing
- The time data error should be fixed in 1.0.3

Thanks for the input!

**Reply by MickeyM007** [reply]
Thanks for the reply!! If I remember correctly, the script did continue after the "Initialization of the cloud." message.

Edit: v1.0.3 works fine so far.. Average speed now=15.1x!
Good job! I'm happy!

**Reply by ghost** [reply]
I stopped developing my script but since you're continuing development, I'll post this here.

[Highlight_Upload_Script.txt](https://github.com/MacielG1/VodRecovery/files/14565924/Highlight_Upload_Script.txt)

The text file I attached retrieves the M3U8 link for subscriber-only highlights/uploads. I did not check if you already have this functionality in your script; however, I don't see it in the README, so I will assume not. The approach may not be the most efficient, but it works and can be used to further enhance the functionality of your fork of my script.

For additional context, the M3U8 link for highlights/uploads does not follow the same format as past broadcasts which makes the current recovery method useless for subscriber-only highlights/uploads specifically. 

@MacielG1

**Reply by MacielG1** [reply]
Thanks! I'll definitely take a look. I initially created this fork for my own use, but since people liked it, I added more features.
I'm curious, what made you stop continuing updating?

**Reply by ghost** [reply]
Twitch's retention policy was a significant factor, but the endless questions about the script were another major issue. It seemed as though people believed they had better insight into the availability of broadcasts than the script itself, which was quite frustrating. As for improvements, beyond implementing proper error handling and optimization changes to enhance recovery speed, there wasn't much else that could be done. These measures might improve the situation to some extent, but they wouldn't fundamentally alter the core functionality.

Diving into the highlight/upload script was a rabbit hole I pursued, and it was surprising that I actually discovered a solution. This was particularly unexpected because it was an issue I had previously looked into but couldn't figure it out.
